.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Java CI with Maven and Release Artifact
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  release:
+    types: [ released ]
+
+permissions:
+  contents: write
+  actions: read
+  checks: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    outputs:
+      version: ${{ steps.extract_version.outputs.version }}
+      artifactId: ${{ steps.extract_artifactId.outputs.artifactId }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B clean install
+
+      - name: Extract project version
+        id: extract_version
+        run: |
+          VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
```

Comment:
I assume this takes the version from the `pom.xml`, right? This would mean that we would usually build a lot of versions with the same number. To be discussed:
- should we include the branch name and/or commit hash in version numbers (in case we are not on the main branch)?
- should we include date/time in the version number?
- do we need version numbers at all if we are not doing a release?